magento/magento2
# [2.1] - Add command to view mview state and queue

- PR for 2.1 = #12050
- PR for 2.2 = #12122
- PR for 2.3 = #12592

### Description

This is similar to the magerun1 command here: https://github.com/netz98/n98-magerun/pull/891

I like the ability to view the mview queue in realtime as its being processed, it can be quite helpful when debugging indexing issues.

This command will actually show how many items are in the list pending processing, as well information from the `mview_state` table.

```
php bin/magento indexer:status:mview
+---------------------------+----------+--------+---------------------+------------+---------+
| ID                        | Mode     | Status | Updated             | Version ID | Backlog |
+---------------------------+----------+--------+---------------------+------------+---------+
| catalog_category_product  | enabled  | idle   | 2017-11-02 10:00:00 | 1          | 0       |
| catalog_product_attribute | enabled  | idle   | 2017-11-02 10:00:00 | 1          | 1       |
| catalog_product_category  | disabled | idle   | 2017-11-02 10:00:00 | 1          | 0       |
| catalog_product_price     | enabled  | idle   | 2017-11-02 10:00:00 | 1          | 0       |
+---------------------------+----------+--------+---------------------+------------+---------+
```

<img width="674" alt="screen shot 2017-11-05 at 20 47 47" src="https://user-images.githubusercontent.com/600190/32419076-3e1d37b4-c26c-11e7-8de0-fdb990cdfea3.png">

### Manual testing scenarios
1. Enable an indexer `php bin/magento indexer:set-mode schedule catalog_product_price`
2. Trigger an action on the product price that will update the `_cl` table
3. Run `php bin/magento indexer:status:mview` to see the backlog increase
4. `magerun2 sys:cron:run indexer_update_all_views` will trigger the mview processes
5. See the backlog decrease 

### Contribution checklist
